lib/intercom-rails/proxy/user.rb

```diff
@@ -30,7 +30,7 @@ def self.current_in_context(search_object)
         potential_user_objects.each do |potential_object|
           begin
             user_proxy = new(search_object.instance_eval(&potential_object), search_object)
-            return user_proxy if user_proxy.valid?
+            return user_proxy if user_proxy.valid? || config.user.as_lead
```

Comment:
@eugeneius Ah! yes yes. I will take it outside the loop and return as a user proxy if ```as_lead``` is present and true :)